src/inmanta/server/services/resourceservice.py

```diff
@@ -1135,29 +1135,28 @@ def _get_query_params(
             attribute_value: Optional[str] = None,
             log_severity: Optional[str] = None,
             limit: Optional[int] = 0,
-        ) -> dict:
+        ) -> dict[str, str]:
             query_params = {
                 "resource_type": resource_type,
                 "agent": agent,
                 "attribute": attribute,
                 "attribute_value": attribute_value,
                 "log_severity": log_severity,
-                "limit": limit,
+                "limit": str(limit) if limit else None,
             }
-            query_params = {param_key: param_value for param_key, param_value in query_params.items() if param_value}
-            return query_params
+            return {param_key: param_value for param_key, param_value in query_params.items() if param_value is not None}
 
         if limit and resource_action_dtos:
             base_url = "/api/v2/resource_actions"
             common_query_params = _get_query_params(resource_type, agent, attribute, attribute_value, log_severity, limit)
             # Next is always earlier with regards to 'started' time
             next_params = common_query_params.copy()
-            next_params["last_timestamp"] = resource_action_dtos[-1].started
-            next_params["action_id"] = resource_action_dtos[-1].action_id
+            next_params["last_timestamp"] = resource_action_dtos[-1].started.isoformat(timespec="microseconds")
```

Comment:
@sanderr these gave warnings which I think were proper errors. I tried to fix them but it would be good if you double check this

Before this was an implicit str() which I think gives the wrong timestamp

Reply:
I think so, but I'm not very familiar with this flow. But from the timestamp perspective, yes. It might be better to call `inmanta.util.datetime_iso_format` though.

Reply:
done